packages/app/blockissuer/blockfactory/factory.go

```diff
@@ -113,6 +113,7 @@ func (f *Factory) createBlockWithPayload(p payload.Payload, references models.Pa
 		return nil, errors.Wrap(err, "signing failed")
 	}
 	block.SetSignature(signature)
+	block.InvalidateBytesCache()
```

Comment:
This is quite ugly. You form a block, then you sign it by computing the block's content hash and therefore you are hashing the block's bytes without yet its signature. This results in the block's bytecache to be the serialized version of the block without its signature.
I think the `sign` method should invalidate the bytes cache, not this method. Put a comment in there as an explaination.

Reply:
Yeah, we talked about it and I think it makes more sense to put it after we actually modify the object, rather than after generating the signature.